src/oom/oomd-manager.c

```diff
@@ -244,11 +252,16 @@ int manager_new(Manager **ret) {
         return 0;
 }
 
-int manager_start(Manager *m, bool dry_run) {
+int manager_start(Manager *m, ResourceMetricMask settings, bool dry_run) {
         int r;
 
         assert(m);
 
+        if (FLAGS_SET(settings, RESOURCE_METRIC_MASK_UNSET))
+                m->detection_mode = RESOURCE_METRIC_MASK_MEM_PRESSURE;
```

Comment:
Use SET_FLAG maybe?

Reply:
I think this with its corresponding else statement might be slightly too complex for SET_FLAG.